src/org/org.ts

```diff
@@ -227,6 +261,62 @@ export class Org extends AsyncOptionalCreatable<Org.Options> {
     });
   }
 
+  /**
+   * Refresh (update) a sandbox from a production org.
+   * 'this' needs to be a production org with sandbox licenses available
+   *
+   * @param sandboxInfo SandboxInfo to update the sandbox with
+   * @param options Wait: The amount of time to wait before timing out, Interval: The time interval between polling
+   */
+  public async refreshSandbox(
+    sandboxInfo: SandboxInfo,
+    options: { wait?: Duration; interval?: Duration; async?: boolean } = {
+      wait: Duration.minutes(6),
+      async: false,
+      interval: Duration.seconds(30),
+    }
+  ): Promise<SandboxProcessObject> {
+    this.logger.debug(sandboxInfo, 'RefreshSandbox called with SandboxInfo');
+    const refreshResult = await this.connection.tooling.update('SandboxInfo', sandboxInfo);
+    this.logger.debug(refreshResult, 'Return from calling tooling.update');
+
+    if (!refreshResult.success) {
+      throw messages.createError('sandboxInfoRefreshFailed', [JSON.stringify(refreshResult)]);
```

Comment:
is `refreshResult` guaranteed to be JSON? Just wondering if we get something unexpected back from the server (like HTML) then it would be an ugly message for the user.

Reply:
Good question.  As long as this uses the jsforce version that catches unexpected HTML responses to SOAP requests it should throw a much better error.